cpp/src/parquet/arrow/writer.h

```diff
@@ -91,10 +91,21 @@ class PARQUET_EXPORT FileWriter {
 
   /// \brief Write a Table to Parquet.
   ///
+  /// If `use_buffering` is false, then any pending row group is closed
+  /// at the beginning and at the end of this call.
+  /// If `use_buffering` is true, this function reuses an existing
+  /// buffered row group until the chunk size is met, and leaves
+  /// the last row group open for further writes.
+  /// It is recommended to set `use_buffering` to true to minimize
+  /// the number of row groups, especially when calling `WriteTable`
+  /// with small tables.
+  ///
   /// \param table Arrow table to write.
   /// \param chunk_size maximum number of rows to write per row group.
-  virtual ::arrow::Status WriteTable(
-      const ::arrow::Table& table, int64_t chunk_size = DEFAULT_MAX_ROW_GROUP_LENGTH) = 0;
+  /// \param use_buffering Whether to potentially buffer data.
+  virtual ::arrow::Status WriteTable(const ::arrow::Table& table,
```

Comment:
How about adding a separate `WriteBufferedTable(const ::arrow::Table& table)` function? 

My rationales are:
- `chunk_size` would be confusing since the `WriterProperties::max_row_group_length()` serves the same purpose. 
- Adding a new `use_buffering` parameter complicates the handling of `WriteTable` to deal with previous state.
- In most cases, we don't expect users to mix calls of `use_buffering=true` and `use_buffering=false`.

Reply:
IIUC, the use case of `WriteTable()` and `WriteRecordBatch()` should be:
- A single RecordBatch object should not be large, so `WriteRecordBatch()` is implemented to write in the buffered mode to avoid small row groups.
- A single Table usually holds a large number of records, so it makes sense to see `WriteTable()` breaks the table into row groups and does not use the buffered mode.

The current PR is addressing the issue when a Table object is small which should copy the same behavior of `WriteRecordBatch`.

Reply:
Yes we can. However, this is mentioned here: https://github.com/apache/arrow/pull/36286#issuecomment-1611332862

And I think that `RecordBatch` reuses Array, `Table` may uses `ChunkedArray`, so supporting a `WriteTable` with buffered is ok here.

Reply:
Sorry that I have missed the discussion. Since the conclusion is to extend the `WriteTable`, I will resolve this comment.